corehq/apps/locations/resources/v0_6.py

```diff
@@ -35,4 +39,68 @@ def dehydrate(self, bundle):
             bundle.data['parent_location_id'] = ''
         bundle.data['location_type_name'] = bundle.obj.location_type.name
         bundle.data['location_type_code'] = bundle.obj.location_type.code
+        bundle.data['location_data'] = bundle.obj.metadata
+        return bundle
+
+    def obj_create(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        if 'name' not in bundle.data:
+            raise BadRequest("'name' is a required field.")
+        if SQLLocation.objects.filter(domain=domain, site_code=bundle.data['site_code']).exists():
+            raise BadRequest("Location on domain with site code already exists.")
```

Comment:
the UI also has validation that names are unique among sibling locations:
https://github.com/dimagi/commcare-hq/blob/bca2d0632f95bbd9bf6601db00dfa5d90d032ec3/corehq/apps/locations/forms.py#L218-L238
That should probably be enforced with a `unique_together` DB constraint, but at present I think it's just code.